oshi-core/src/main/java/oshi/jna/platform/linux/Libc.java

```diff
@@ -51,15 +58,107 @@
         public NativeLong totalhigh; // Total high memory size
         public NativeLong freehigh; // Available high memory size
         public int mem_unit; // Memory unit size in bytes
-        public byte[] _f = new byte[8]; // Won't be written for 64-bit systems
+        // Padding to 64 bytes
+        public byte[] _f = new byte[PADDING_SIZE];
+
+        /*
+         * getFieldList and getFieldOrder are overridden because PADDING_SIZE
+         * might be 0 - that is a GCC only extension and not supported by JNA
+         * 
+         * The dummy field at the end of the structure is just padding and so if
+         * the field is the zero length array, it is stripped from the fields
+         * and field order.
+         */
+        @Override
+        protected List<Field> getFieldList() {
+            List<Field> fields = new ArrayList<>(super.getFieldList());
+            if (PADDING_SIZE == 0) {
+                Iterator<Field> fieldIterator = fields.iterator();
+                while (fieldIterator.hasNext()) {
+                    Field field = fieldIterator.next();
+                    if ("_f".equals(field.getName())) {
+                        fieldIterator.remove();
+                    }
+                }
+            }
+            return fields;
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            List<String> fieldOrder = new ArrayList<>(super.getFieldOrder());
+            if (PADDING_SIZE == 0) {
+                fieldOrder.remove("_f");
+            }
+            return fieldOrder;
+        }
+    }
+
+    @FieldOrder({ "fsBlockSize", "fsFragmentSize", "fsSizeInBlocks", "fsBlocksFree", "fsBlocksFreeUnpriv",
+            "fsTotalInodeCount", "fsFreeInodeCount", "fsFreeInodeCountUnpriv", "fsId", "_fPad32bit", "fsMountFlags",
+            "fsMaxFilenameLength", "_fSpare" })
+    class Statvfs extends Structure {
+        private static final int PADDING_SIZE = 8 / NativeLong.SIZE - 1;
+
+        public NativeLong fsBlockSize;
+        public NativeLong fsFragmentSize;
+        public NativeLong fsSizeInBlocks;
+        public NativeLong fsBlocksFree;
+        public NativeLong fsBlocksFreeUnpriv;
+        public NativeLong fsTotalInodeCount;
+        public NativeLong fsFreeInodeCount;
+        public NativeLong fsFreeInodeCountUnpriv;
+        public NativeLong fsId;
+        public int[] _fPad32bit = new int[PADDING_SIZE]; // 0 on 64-bit systems
```

Comment:
So I'm looking through docs everywhere and can't find why this padding exists. Can you point me to the source you used to determine this structure?

Reply:
Ah, nevermind. [Found it here. ](https://code.woboq.org/gcc/include/bits/statvfs.h.html) We should name it `_f_unused` though.